lib/model/message.dart

```diff
@@ -56,7 +61,10 @@ class MessageStoreImpl with MessageStore {
   }
 
   void dispose() {
-    for (final view in _messageListViews) {
+    // When a MessageListView is disposed, it removes itself from the set
+    // `MessageStoreImpl._messageListViews`. Instead of iterating on that set,
+    // iterate on a copy, to avoid concurrent modifications.
+    for (final view in _messageListViews.toList()) {
```

Comment:
nit in commit message:

    This fixes a bug caused by MessageListView instances unregistering themselves
    from the collection containing these instances, while we are iterating the exact
    same set when disposing the message store.
    
lines too long for prose; from https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#formatting-guidelines :
> Your commit message should be line-wrapped to about 68 characters per line, but no more than 70, so that your commit message will be easy to read in `git log` in a normal terminal.

These lines are as long as 80 columns.

See also the tip there about configuring Git and your editor.